### ISSUE TITLE
Add support for distributed tracing

### ISSUE THREAD

**Issue by spencergibb** [spring-cloud/spring-cloud-commons#11]
Putting in s-c-commons because this might warrant a new sub-project.

From this twitter conversation: https://twitter.com/StephenDillon7/status/565132369385254912

I could see some easy opportunities for auto-configuration and integration with other s-c projects.  Possible opportunity to send the data into spring XD.

### Twitter Zipkin
http://twitter.github.io/zipkin

> Zipkin is a distributed tracing system that helps us gather timing data for all the disparate services at Twitter. It manages both the collection and lookup of this data through a Collector and a Query service. We closely modelled Zipkin after the [Google Dapper paper](http://research.google.com/pubs/pub36356.html). Follow [@ZipkinProject](http://twitter.com/ZipkinProject) for updates.

> Collecting traces helps developers gain deeper knowledge about how certain requests perform in a distributed system. Let’s say we’re having problems with user requests timing out. We can look up traced requests that timed out and display it in the web UI. We’ll be able to quickly find the service responsible for adding the unexpected response time. If the service has been annotated adequately we can also find out where in that service the issue is happening.

### Brave
https://github.com/kristofa/brave

> Java distributed tracing implementation compatible with Zipkin.

Basically, alternative client code not dependent on scala.  Has some spring support.

**Comment by shakuzen** [comment]
+1

**Comment by torsten-liermann** [comment]
+1

**Comment by dgomesbr** [comment]
hey @spencergibb any updates on this? 

got here from https://taidevcouk.wordpress.com/2014/05/26/implementing-correlation-ids-in-spring-boot/ and other posts implementing filters and plagging the code with get from request, put on request.

**Comment by dgomesbr** [comment]
cc @kelapure

**Comment by spencergibb** [comment]
@dgomesbr doing some work on it in [spring-cloud-sleuth](https://github.com/spring-cloud-incubator/spring-cloud-sleuth).

**Comment by dgomesbr** [comment]
Thanks for the promptly response @spencergibb, gonna follow over there and see in what could I help with.

**Comment by spencergibb** [comment]
Closing in favor of https://github.com/spring-cloud/spring-cloud-sleuth
